packages/backend/src/payment-method/ilp/connector/core/middleware/expire.ts

```diff
@@ -14,13 +14,29 @@ export function createOutgoingExpireMiddleware() {
   ): Promise<void> => {
     const { expiresAt } = request.prepare
     const duration = expiresAt.getTime() - Date.now()
-    const timeout = setTimeout(() => {
-      logger.debug({ request }, 'packet expired')
-      throw new TransferTimedOutError('packet expired.')
-    }, duration)
 
-    await next().finally(() => {
-      clearTimeout(timeout)
-    })
+    let timeoutId
+    const timeout = async (): Promise<never> =>
+      new Promise((_, reject) => {
+        timeoutId = setTimeout(() => {
+          reject(new TransferTimedOutError('packet expired.'))
+        }, duration)
+      })
+
+    try {
+      if (duration <= 0) {
+        throw new TransferTimedOutError('packet expired.')
+      }
+      await Promise.race([next(), timeout()])
+    } catch (error) {
+      if (error instanceof TransferTimedOutError) {
+        logger.debug({ request }, 'packet expired')
```

Comment:
The `request` object in the log is large:

```
{"level":20,"time":1732705456994,"pid":175,"hostname":"cloud-nine-wallet-backend","service":"ConnectorService","request":{"prepare":{"_expiresAtChanged":true,"_amountChanged":true,"_prepare":{"amount":"550","executionCondition":{"type":"Buffer","data":[66,218,235,58,214,143,11,42,36,206,83,237,136,184,6,92,93,30,68,136,89,176,251,71,57,183,68,12,36,150,40,145]},"expiresAt":"2024-11-27T11:04:14.873Z","destination":"test.happy-life-bank.5XSMbcSG-8_-ZrXTyzSZPCiwJcKK-2XynhVyKBszxiMIxgqz6slbbPIMvl74YzdDhrRl31lWqcHrQ0lhnFV89rtXJrfY89Y","data":{"type":"Buffer","data":[211,160,155,236,173,57,110,84,164,146,242,42,231,35,38,25,172,246,63,96,198,204,7,234,50,198,66,141,134,101,103,20,238,46,169,130,203,167,162,83,52,164,221,230,49,143,183,136,209,184,92]}},"_amount":"550","_originalAmount":"550","_originalExpiresAt":"2024-11-27T11:04:15.873Z"},"rawPrepare":{"type":"Buffer","data":[12,129,226,0,0,0,0,0,0,2,38,50,48,50,52,49,49,50,55,49,49,48,52,49,52,56,55,51,66,218,235,58,214,143,11,42,36,206,83,237,136,184,6,92,93,30,68,136,89,176,251,71,57,183,68,12,36,150,40,145,116,116,101,115,116,46,104,97,112,112,121,45,108,105,102,101,45,98,97,110,107,46,53,88,83,77,98,99,83,71,45,56,95,45,90,114,88,84,121,122,83,90,80,67,105,119,74,99,75,75,45,50,88,121,110,104,86,121,75,66,115,122,120,105,77,73,120,103,113,122,54,115,108,98,98,80,73,77,118,108,55,52,89,122,100,68,104,114,82,108,51,49,108,87,113,99,72,114,81,48,108,104,110,70,86,56,57,114,116,88,74,114,102,89,56,57,89,51,211,160,155,236,173,57,110,84,164,146,242,42,231,35,38,25,172,246,63,96,198,204,7,234,50,198,66,141,134,101,103,20,238,46,169,130,203,167,162,83,52,164,221,230,49,143,183,136,209,184,92]}},"msg":"packet expired"}
```
maybe we can narrow it down to the destination, and the packet's expiry